### DISCUSSION TITLE
are there any python scripts able to draw the network topology based on the shadow.log or config.xml?

### DISCUSSION THREAD

**Discussion by GitZinger** [shadow/shadow#769] [Q&A] [answered]
**State your question concisely**
are there any python scripts able to draw the network topology based on the shadow.log or config.xml?
**Describe your current understanding of Shadow as it relates to your question**
it is hard to read the topology right now. 
**Describe which parts you are unsure about**
any visualization ways based on those nodes and edges?

**Comment by robgjansen** [comment, marked as answer]
Sorry, we have not developed such visualization tools. However, the network topology is in the standard `graphml` format, and there are many tools out there on the Internet that can visualize graphml files. For example, https://gephi.org/
